AST diff crash related to ParamSpec

I encountered this crash when using mypy daemon:
```
Traceback (most recent call last):
  File "mypy/dmypy_server.py", line 229, in serve
  File "mypy/dmypy_server.py", line 272, in run_command
  File "mypy/dmypy_server.py", line 371, in cmd_recheck
  File "mypy/dmypy_server.py", line 528, in fine_grained_increment
  File "mypy/server/update.py", line 245, in update
  File "mypy/server/update.py", line 328, in update_one
  File "mypy/server/update.py", line 414, in update_module
  File "mypy/server/update.py", line 834, in propagate_changes_using_dependencies
  File "mypy/server/update.py", line 924, in reprocess_nodes
  File "mypy/server/astdiff.py", line 160, in snapshot_symbol_table
  File "mypy/server/astdiff.py", line 226, in snapshot_definition
AssertionError: <class 'mypy.nodes.ParamSpecExpr'>
```

It looks like `astdiff.py` hasn't been updated to deal with `ParamSpecExpr`.

--- a/mypy/server/astdiff.py
+++ b/mypy/server/astdiff.py
@@ -54,7 +54,7 @@ from typing import Set, Dict, Tuple, Optional, Sequence, Union
 
 from mypy.nodes import (
     SymbolTable, TypeInfo, Var, SymbolNode, Decorator, TypeVarExpr, TypeAlias,
-    FuncBase, OverloadedFuncDef, FuncItem, MypyFile, UNBOUND_IMPORTED
+    FuncBase, OverloadedFuncDef, FuncItem, MypyFile, ParamSpecExpr, UNBOUND_IMPORTED
 )
 from mypy.types import (
     Type, TypeVisitor, UnboundType, AnyType, NoneType, UninhabitedType,
@@ -151,6 +151,10 @@ def snapshot_symbol_table(name_prefix: str, table: SymbolTable) -> Dict[str, Sna
                             node.normalized,
                             node.no_args,
                             snapshot_optional_type(node.target))
+        elif isinstance(node, ParamSpecExpr):
+            result[name] = ('ParamSpec',
+                            node.variance,
+                            snapshot_type(node.upper_bound))
         else:
             assert symbol.kind != UNBOUND_IMPORTED
             if node and get_prefix(node.fullname) != name_prefix:

--- a/test-data/unit/diff.test
+++ b/test-data/unit/diff.test
@@ -1470,3 +1470,17 @@ x: Union[Callable[[Arg(int, 'y')], None],
 [builtins fixtures/tuple.pyi]
 [out]
 __main__.x
+
+[case testChangeParamSpec]
+from typing import ParamSpec, TypeVar
+A = ParamSpec('A')
+B = ParamSpec('B')
+C = TypeVar('C')
+[file next.py]
+from typing import ParamSpec, TypeVar
+A = ParamSpec('A')
+B = TypeVar('B')
+C = ParamSpec('C')
+[out]
+__main__.B
+__main__.C

